src/models/Configuration.php

```diff
@@ -132,6 +133,39 @@ private static function configurationFromRow(
     );
   }
 
+  public static function genFacebookOAuthSettingsExists(): bool {
+    $settings_file = '../../settings.ini';
+    $config = parse_ini_file($settings_file);
+
+    if ((array_key_exists('FACEBOOK_OAUTH_APP_ID', $config) === true) &&
+        (array_key_exists('FACEBOOK_OAUTH_APP_SECRET', $config) === true)) {
+      return true;
+    }
+    return false;
+  }
+
+  public static function genFacebookOAuthSettingsAppId(): string {
+    $settings_file = '../../settings.ini';
+    $config = parse_ini_file($settings_file);
```

Comment:
For both this and the existing google things, we shouldn't have this repeated as often; split the ini parsing to a function, cache it, and use TypeAssert to assert it matches your expected safe to get typesafety

Reply:
Great idea.  We can certainly cache this and not repeatedly parse the API keys from disk.